README.md

```diff
@@ -67,8 +68,28 @@ class MyCharm(CharmBase):
         # Handle the event here.
 ```
 
-Every standard event in Juju may be observed that way, and you can also easily
-define your own events in your custom types.
+In the above example, we are telling the framework we want to observe the
+`self.on.start` event and that we want the `self.on_start` handler to respond
+to that event. Every standard event in Juju may be observed that way, and you
+can also easily define your own events in your custom types.
+
+Alternatively, if you name your handler with the scheme `on_<event-name>` then
+explicitly referencing said handler is optional and you can just provide the
+handler's parent observer. For example:
```

Comment:
I think I do prefer the approach that @mthaddon did in
https://github.com/canonical/operator/pull/115

Which is to start with the simpler syntax, and then have a comment after about how the convention of exact names can be change to an explicit function. I do think we'll want to have that relatively early in the documentation, so that people don't start having:

```
def __init__(...):
  self.framework.observe(self.on.start, self)
  self.framework.observe(self,on.config_changed, self)
...
def on_start(self, event):
  self.common_handler(event)

def on_config_changed(self, event):
  self.common_handler(event)
```